### ISSUE TITLE
New launcher script does not allow the execution of legacy version with Java 7

### ISSUE THREAD

**Issue by senthil10** [nextflow-io/nextflow#519]
**ISSUE:**

The new version *(`0.26`)* requires `JAVA` version be *>=8*. So even if we specify earlier version while download it still asks for `JAVA 8` or higher. For example

```
export NXF_VER="0.25.7"
wget -qO- get.nextflow.io | bash
```

The above command downloads the binary for version `0.25.7`, but still the original executable binary file (`nextflow`) is the recent one, where it is hardcoded like following (line 291-295)

```
 if [[ ! $JAVA_VER =~ ^1.8 && ! $JAVA_VER =~ ^9 ]]; then
      echo_red "ERROR: Cannot find Java or it's a wrong version -- please make sure that Java 8 is    installed"
      echo_yellow "NOTE: Nextflow is trying to use the Java VM defined by the following environment   variables:\n JAVA_CMD: $JAVA_CMD\n JAVA_HOME: $JAVA_HOME\n"
      exit 1
  fi
 ```

Since we install `nextflow` in our cluster with all our other tools, we cannot use `JAVA 8+` as one of other crucial tool needs `JAVA 7`. 

**WORK AROUND:**

So for now we decided to use nextflow `0.25.7` and I was able download the right version (of `nextflow` file) by

```
wget -qO- https://www.nextflow.io/releases/v0.25.7/nextflow
```

I found the above link from `get()/install()` function in the `nextflow` source code. I was not able to find any documentation regarding that.

**CONCLUSION:**

Not sure if this was intended behaviour that the default `nextflow` executable files requires `JAVA 8+` even though `NXF_VER` is set to earlier version.

If it is, would great to update documentation on how to download and install older version properly (with proper support to `JAVA 7+`)

If not, would better to evaluate the `JAVA` requirement based on `NXF_VER` rather than hardcoding it.

Thank you!!
/Senthil

**Comment by pditommaso** [comment]
I've uploaded a pushed a patch you may want to review/test it.

**Comment by pditommaso** [comment]
Looks much better now https://github.com/nextflow-io/nextflow/commit/34f9190f9541c4da3047e1f4494f7a3c493f6648.

**Comment by pditommaso** [comment]
Included in version `0.26.1-SNAPSHOT build 4732`
